src/main/scala/thurloe/notification/NotificationMonitor.scala

```diff
@@ -194,12 +194,29 @@ class NotificationMonitorActor(val pollInterval: FiniteDuration,
 
   def lookupShouldNotify(notification: Notification): Future[Boolean] =
     notification match {
+      //For workspace notifications, there are three tiers of preferences to check:
+      // 1. has the user disabled *all* notifications for their account?
+      //     key format: notifications.off
+      // 2. has the user disabled the notification at the type-level?
+      //     key format: notifications/<notification class name>
+      // 3. has the user disabled the notification for the specified workspace?
+      //     key format: notifications/<notification class name>/<workspace namespace>/<workspace name>
+      case wsNotification: WorkspaceNotification =>
+        val baseKey = s"notifications/${wsNotification.getClass.getSimpleName}"
+        //Check #1
+        booleanLookup(wsNotification.recipientUserId, NotificationMonitor.notificationsOffKey, false) flatMap {
+          case false =>
+            booleanLookup(wsNotification.recipientUserId, notification.key, true) flatMap {
+              case false => Future.successful(false)
+              case true  => booleanLookup(wsNotification.recipientUserId, baseKey, true)
+            }
```

Comment:
If I'm reading this right, this checks `notifications/<notification class name>/<workspace namespace>/<workspace name>` before  `notifications/<notification class name>`? I guess it doesn't matter, but I would have expected the checks to follow the order in the comment.

Reply:
Whoops yeah, ideally it should follow the order in the comment. I'll update that for sanity